src/main/react/app/routes/builder/context/builder-context.tsx

```diff
@@ -36,24 +90,61 @@ export default function BuilderContext({ onClose }: Readonly<{ onClose: () => vo
             className="w-full rounded-full border border-gray-200 bg-gray-50 py-1 pr-4 pl-7"
             type="search"
             placeholder="Search"
+            value={searchTerm}
+            onChange={(e) => setSearchTerm(e.target.value)}
           />
         </div>
       </div>
       <div className="h-full overflow-y-auto">
         <ul className="p-4">
           {isLoading && <li>Loading...</li>}
           {error && <li>Error: {error}</li>}
+          {!isLoading && Object.keys(elementsToRender).length === 0 && (
+            <li className="text-gray-500 italic">No results found.</li>
+          )}
           {!isLoading &&
-            Object.entries(frankDocRaw?.elements).map(([key, value]: [string, any]) => (
-              <li
-                className="m-2 cursor-move rounded border border-gray-400 p-4"
-                key={value.name}
-                draggable
-                onDragStart={onDragStart(value)}
-              >
-                {value.name}
-              </li>
-            ))}
+            Object.entries(elementsToRender)
+              .sort(([typeA], [typeB]) => typeA.localeCompare(typeB))
+              .map(([type, items]) => {
+                const isExpanded = searchTerm ? true : (expandedGroups[type] ?? true)
```

Comment:
Does this logic really need to be like this?
Wouldn't this produce the same result?
```suggestion
                const isExpanded = expandedGroups[type] ?? searchTerm
```

Is there some reason there should be changed on searchTerm first?

Reply:
The purpose of this code is to auto expand all groups when a user has entered something into the searchfield. In your suggestion the group would not be expanded if the toggle for it is false, even if a user would then enter something in the field. 
I felt like that could make users miss certain elements if they did not think to toggle the specific group where it would be located in

Reply:
Ah, so maybe this then:
```ts
searchTerm || expandedGroups[type] ?? true
```